MeshLib/Properties.h

```diff
@@ -153,21 +150,40 @@ class Properties
 						boost::any_cast<PropertyVector<T> const&>(it->second)
 					);
 			} catch (boost::bad_any_cast const&) {
-				ERR("A property with the specified data type is not available.");
 				return boost::optional<PropertyVector<T> const&>();
 			}
 		} else {
-			ERR("A property with the specified name and MeshItemType is not available.");
+			ERR("A property with the specified name is not available.");
 			return boost::optional<PropertyVector<T> const&>();
 		}
 	}
 
-	void removeProperty(std::string const& name,
-		MeshItemType mesh_item_type)
+	/// Method to get a vector of property values.
+	template <typename T>
+	boost::optional<PropertyVector<T>&>
+	getProperty(std::string const& name)
```

Comment:
sorry i'm not getting why we need this non-const version. does this function change any class variables?